### ISSUE TITLE
processor load: IDB.exe consumes 100% of a core even when doing nothing

### ISSUE THREAD

**Issue by whitneyschmidt** [mono/mono#19951]
_From @thomiel on Wed, 10 Jun 2020 14:48:01 GMT_

### Steps to Reproduce

1. start Visual Studio 2019 (16.6.2)
2. connect to a Mac build agent
3. open activity monitor on Mac
4. sort by % CPU

### Expected Behavior

While idle, process mono-sgen64/IDB.exe should use > 1% of a cpu kernel? 

### Actual Behavior

mono-sgen64/IDB.exe heats up my Mac with 100% utilization of a cpu kernel while no build process is ongoing. Currently I'm use my Mac Mini as a coffee mug warmer when working on iOS projects.

Can't you just add sleep cycles when its idle? Could save some electricity cost and carbon dioxide pumped into the atmosphere.

### Environment

```
Visual Studio Community 2019 for Mac
Version 8.4.8 (build 2)
Installation UUID: f6b80b8d-3056-42ad-a38f-15ec50a80386
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.16.1.25 (issue-7441-d16-3-vsmac / 881172e73)

	Package version: 608000099

Mono Framework MDK
Runtime:
	Mono 6.8.0.99 (2019-10/1182f8cbf5b) (64-bit)
	Package version: 608000099

Roslyn (Language Service)
3.4.0-beta4-19562-05+ff930dec4565e2bc424ad3bf3e22ecb20542c87d

NuGet
Version: 5.3.0.6192

.NET Core SDK
SDK: /usr/local/share/dotnet/sdk/3.1.102/Sdks
SDK-Version: 3.1.102
MSBuild-SDKs: /Library/Frameworks/Mono.framework/Versions/6.8.0/lib/mono/msbuild/Current/bin/Sdks

.NET Core-Runtime
Laufzeit: /usr/local/share/dotnet/dotnet
Laufzeitversionen:
	3.1.2
	2.1.16

Xamarin.Profiler
Version: 1.6.13.11
Speicherort: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode 11.5 (16139)
Build 11E608c

Xamarin.Mac
Version: 6.10.0.21 (Visual Studio Community)
Hash: 02c4b3bdc
Branch: xcode11.3
Build date: 2020-02-18 14:13:31-0500

Xamarin.iOS
Version: 13.18.2.1 (Visual Studio Community)
Hash: 29c4ea731
Branch: d16-6
Build date: 2020-05-26 17:03:05-0400

Xamarin.Android
Nicht installiert

Microsoft Mobile OpenJDK
Java SDK: Nicht gefunden

Android Designer EPL-Code ist hier verfügbar:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 16.4.0.10
Hash: 2c49a7d
Branch: remotes/origin/d16-4
Build date: 2020-02-20 19:25:52 UTC

Android Device Manager
Version: 16.4.0.32
Hash: 7a5cb8b
Branch: remotes/origin/d16-4
Build date: 2020-02-20 19:26:14 UTC

Xamarin Designer
Version: 16.4.0.479
Hash: 074544417
Branch: remotes/origin/d16-4
Build date: 2020-01-22 22:50:22 UTC

Xamarin Inspector
Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

Build Information
Release ID: 804080002
Git revision: 4f35aa7e44fb398379e512d0bfd6f8df8d34b5ac
Build date: 2020-02-27 16:16:52+00
Build branch: release-8.4
Xamarin extensions: 4f35aa7e44fb398379e512d0bfd6f8df8d34b5ac

Operating System
Mac OS X 10.15.4
Darwin 19.4.0 Darwin Kernel Version 19.4.0
    Wed Mar  4 22:28:40 PST 2020
    root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
```

_Copied from original issue xamarin/xamarin-macios#8826_

**Comment by andycted** [comment]
Actually I noticed that sometimes this doesn't happen and the build is quite fast. Instead when cpu is high, build is a crawl...

**Comment by andycted** [comment]
not solved by last update to 16.6.2

**Comment by whitneyschmidt** [comment]
@andycted Thank you for posting your feedback here - future updates to this will be posted here: https://github.com/xamarin/xamarin-macios/issues/8826 Closing this issue since it's not related to the mono team.

The xamarin-macios issue is also the best place to share information and ask questions.
